query/Matcher.java

```diff
@@ -132,43 +135,37 @@ public static Matcher create(Reasoner reasoner, TypeQLMatch query, Context.Query
 
     private FunctionalIterator<ConceptMap> sort(FunctionalIterator<ConceptMap> answers, Sortable.Sorting sorting) {
         // TODO: Replace this temporary implementation of TypeQL Match Sort query with a native sorting traversal
-        Reference.Name var = sorting.var().reference().asName();
+        List<Reference.Name> sortVars = iterate(sorting.vars()).map(var -> var.reference().asName()).toList();
+        Comparator<List<Attribute>> multiComparator = multiComparator(sortVars.size());
         Comparator<ConceptMap> comparator = (answer1, answer2) -> {
-            Attribute att1, att2;
-            try {
-                att1 = answer1.get(var).asAttribute();
-                att2 = answer2.get(var).asAttribute();
-            } catch (TypeDBException e) {
-                if (e.code().isPresent() || e.code().get().equals(INVALID_THING_CASTING.code())) {
-                    throw TypeDBException.of(SORT_VARIABLE_NOT_ATTRIBUTE, var);
-                } else {
-                    throw e;
+            List<Attribute> attributes1 = new ArrayList<>(sortVars.size());
+            List<Attribute> attributes2 = new ArrayList<>(sortVars.size());
+            for (Reference.Name var : sortVars) {
+                try {
+                    attributes1.add(answer1.get(var).asAttribute());
+                    attributes2.add(answer2.get(var).asAttribute());
+                } catch (TypeDBException e) {
+                    if (e.code().isPresent() || e.code().get().equals(INVALID_THING_CASTING.code())) {
+                        throw TypeDBException.of(SORT_VARIABLE_NOT_ATTRIBUTE, var);
+                    } else {
+                        throw e;
+                    }
                 }
             }
-
-            if (!att1.getType().getValueType().comparables().contains(att2.getType().getValueType())) {
-                throw TypeDBException.of(SORT_ATTRIBUTE_NOT_COMPARABLE, var);
-            }
-            if (att1.isString()) {
-                return att1.asString().getValue().compareToIgnoreCase(att2.asString().getValue());
-            } else if (att1.isBoolean()) {
-                return att1.asBoolean().getValue().compareTo(att2.asBoolean().getValue());
-            } else if (att1.isLong() && att2.isLong()) {
-                return att1.asLong().getValue().compareTo(att2.asLong().getValue());
-            } else if (att1.isDouble() || att2.isDouble()) {
-                Double double1 = att1.isLong() ? att1.asLong().getValue() : att1.asDouble().getValue();
-                Double double2 = att2.isLong() ? att2.asLong().getValue() : att2.asDouble().getValue();
-                return double1.compareTo(double2);
-            } else if (att1.isDateTime()) {
-                return (att1.asDateTime().getValue()).compareTo(att2.asDateTime().getValue());
-            } else {
-                throw TypeDBException.of(ILLEGAL_STATE);
-            }
```

Comment:
this has been moved into the `Attribute` class, which in turn gets comparators to use from the `ValueType` and `Encoding` classes